arroyo-worker/src/engine.rs

```diff
@@ -215,10 +220,16 @@ impl<K: Key, T: Data> Collector<K, T> {
 
             for (i, out_node_qs) in self.out_qs.iter().enumerate() {
                 let idx = out_idx(&key, out_node_qs.len());
-                self.tx_queue_size[i][idx]
+                self.tx_queue_rem_gauges[i][idx]
                     .iter()
                     .for_each(|c| c.set(self.out_qs[i][idx].tx.capacity() as i64));
 
+                self.backpressure_gauges[i][idx]
+                    .iter()
+                    .for_each(|c| c.set(
+                        ((1.0 - self.out_qs[i][idx].tx.capacity() as f64 / QUEUE_SIZE as f64) * 100.0) as i64)
```

Comment:
I'm calculating the _percentage full_ of the queue and multiplying it by 100 so we can use the `IntegerGauge`.  Seems like a lot of type converting, @mwylde what do you think of this?

Reply:
What about just reporting queue size, and then doing the math on the query side?

Reply:
Currently in `metrics.rs` the `get_query` function hardcodes a simple `rate` query of a single metric. Are you suggesting modifying that to be able to do a more complex promql query that combines two metrics?

Or are you suggesting doing the calculation in the frontend?

Reply:
I did the former.